### PR TITLE
[docs] Add "Integrating a Go project" documentation page (#2714).

### PR DESCRIPTION
**Pull request by Dor1s**

### CODE REVIEW

**Comment by Dor1s** [comment]
Ended up adding a grandchild page with Go specifics, and left the main New Project Guide almost untouched, as it's pretty agnostic and applicable to Go projects as well.

File: docs/getting-started/new-project-guide/go_lang.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+to the general
+[Setting up a new project]({{ site.baseurl }}/getting-started/new-project-guide/)
+process. The key specifics of integrating a Go project are outlined below.
+
+## Go-fuzz support
+
+OSS-Fuzz supports **go-fuzz** in the
+[libFuzzer compatible mode](https://github.com/dvyukov/go-fuzz#libfuzzer-support)
+only. In that mode, fuzz targets for Go code are using libFuzzer fuzzing engine
+with the native Go coverage instrumentation. The binaries compiled in that mode
```
Comment (oliverchang) [inline docs/getting-started/new-project-guide/go_lang.md]: nit: maybe reword as

"In that mode, fuzz targets for Go use the libFuzzer engine with native Go coverage instrumentation. Binaries compiled in this mode provide the same libFuzzer command line interface as non-Go fuzz targets."

Comment (Dor1s) [reply docs/getting-started/new-project-guide/go_lang.md]: this sounds better, thanks!
